system/codesigner/gpgcodesigner.cpp

```diff
@@ -0,0 +1,250 @@
+#include "jlib.hpp"
+#include "jexcept.hpp"
+#include "jlog.hpp"
+#include "gpgcodesigner.hpp"
+#include "atomic"
+
+class GpgCodeSigner : implements ICodeSigner
+{
+    virtual bool sign(const char * text, const char * userId, const char * passphrase, StringBuffer & errmsg, StringBuffer & signedText) override;
```

Comment:
a slightly pedantic comment which you are welcome to ignore:
these functions should really be const since logically they do not modify the contents of the code signing object, and isGpgv1 and isInitialised should be mutable.

Reply:
initGpg is non-const.  If sign is const then it wont be able to initGpg.  (In another PR, I use initGpg to also import keys from secrets so that keys are not unnecessarily imported, slowing down compilation.)